### PR TITLE
Utilize k3s for e2e testing #678

### LINKED ISSUE #678: Utilize k3s for e2e testing
**Is this a BUG REPORT or FEATURE REQUEST?**:
More like a FEATURE REQUEST.

**What happened**:
Current e2e tests don't run all existing examples.

**What you expected to happen**:
It should be possible to run all existing examples as an e2e test.

There is a make target right now that runs unit tests. It'll be good have a make target, say `make e2e-test` that runs the above.

### PR DESCRIPTION
**Pull request by sarabala1979**
This PR will fix #678 .
Added E2E step in CI to run the E2E testing using k3s.
Implementation Details:
1. Add E2E test case config file. It will contain Testname, workflow path and expected status
2. Add E2E go testcase which will read e2e config file and execute  testcases

### CODE REVIEW

**Comment by xianlubird** [comment]
Can we also run this e2e test in local env.

Just like we provide a kubeConfig file which access to some k8s cluster ,and e2e test will throw these workflows to k8s cluster through apiServer

**Comment by sarabala1979** [comment]
you can run locally in two versions.
**Run with un-push changes:**
go test -timeout 45m --tags=e2e ./test/e2e

Run from repo:
argo submit .argo-ci/ci.yaml

File: r.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+kind: Role
```
Comment (jessesuen) [inline r.yaml]: Accidental checkin?

File: test.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+apiVersion: argoproj.io/v1alpha1
```
Comment (jessesuen) [inline test.yaml]: Accidental checkin?

**Comment by alexec** [comment]
Closing as superseded.

**Pull request closed without merging**